lib/ansible/utils/plugin_docs.py

```diff
@@ -42,14 +42,15 @@ def merge_fragment(target, source):
 
 def _process_versions_and_dates(fragment, is_module, return_docs, callback):
     def process_deprecation(deprecation, top_level=False):
+        collection_name = 'removed_from_collection' if top_level else 'collection_name'
```

Comment:
What's this change for? Currently deprecated `env` sources display `removed_from_collection: ansible.builtin` and this will change it to `collection_name: ansible.builtin`

Reply:
The code which shows the deprecation message in Ansible uses `collection_name`, not `removed_from_collection`: https://github.com/ansible/ansible/blob/devel/lib/ansible/cli/__init__.py#L100

The deprecation entries are put into `C.config.DEPRECATED` in https://github.com/ansible/ansible/blob/devel/lib/ansible/config/manager.py#L403 and https://github.com/ansible/ansible/blob/devel/lib/ansible/config/manager.py#L488 and https://github.com/ansible/ansible/blob/devel/lib/ansible/config/manager.py#L522

Reply:
Should plugins be standardized with modules? Looks like modules support 'removed_from_collection' for options https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/parameters.py#L144

Reply:
Though that's in argument_spec, not in `DOCUMENTATION`. I'm fine with changing that, though, though it could be a bit of a breaking change.

Reply:
Also, the other fields are already named differently: for plugins, they are called `version` and `date`, while for argument_spec, they are called `removed_in_version` and `removed_at_date`.

Reply:
The plugin docs double as the argspec, so I wasn't sure if we wanted consistency. But this looks good to me as-is.

Reply:
I think consistency would be nice, but it requires some more work to be backwards compatible. It's probably better to do separately.